modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/search/engine/adapter/search/SearchSearchRequestAssemblerImpl.java

```diff
@@ -153,6 +171,13 @@ private void _setFetchSource(
 				searchSearchRequest.getFetchSourceIncludes(),
 				searchSearchRequest.getFetchSourceExcludes());
 		}
+		else {
+			if (ArrayUtil.isEmpty(
```

Comment:
im curious about this... seems that you could also make the argument that we can also set fetchSource(false) if **_isNotEmpty_**(searchSearchRequest.getSelectedFieldNames()) (or maybe you did make this argument in your comment in the LPD and you accidentally used isEmpty instead of isNotEmpty?)

in other words, now that we are using `fields`, fetchSource can potentially just default to false now always, and only turn it on if searchSearchRequest asks for it?

Reply:
If selected fields is empty, assembler sets the fields to '*', which then, with `fields` retrieval triggers retrieving the source automatically. Maybe it could be set to false by default, but if `fields` is not empty, the source is not retrieved by default anyway.

Reply:
ok right, i got that confused, empty `selectedFieldNames` == `"fields": ["*"]`, which still returns `_source`, so `fetchSource(false)` is needed

at the same time, populated `selectedFieldNames` == `"fields": ["xyz"]`, which still returns `_source`, so `fetchSource(false)` is needed. i think this contradicts the last comment you made? >"but if fields is not empty, the source is not retrieved by default anyway."

Reply:
i think it was when stored_fields is not empty, the source is not retrieved by default

Reply:
Right, i made a testing mistake. Source seems to be allways included by default with `fields`. Putting the default to false.